src/api/status/src/server.js

```diff
@@ -77,17 +77,17 @@ service.router.get(process.env.PATH_PREFIX || '/', async (req, res) => {
   let totalFeeds;
 
   try {
-    [telescope, satellite, totalPost, totalFeeds] = await Promise.all([
-      getGitHubData('Seneca-CDOT', 'telescope'),
-      getGitHubData('Seneca-CDOT', 'satellite'),
+    [totalPost, totalFeeds, telescope, satellite] = await Promise.all([
```

Comment:
Catching up on your discussion on Slack last night about rate limiting, what I'd do is separate this into a few files.  You want the HTTP route to work with data, but not worry about where it comes from.  That data should either be cached (in Redis) or live from GitHub, and you can combine it all into 1 object.

```js
// a single method returns an object with all the data we need (totalPosts, totalFeeds, etc.)
const telescopeStats = await getTelescopeData();
```

The `getTelescopeData()` function can look like this:

```js
const { redis } = require('@senecacdot/satellite');

const totalPosts = async () => ({
  // Step 1: see if we have this value stored in Redis already
  const cached = await redis.get('github:totalPosts');
  if(cached) {
    return cached;
  }

  // Step 2. get the value from GitHub, cache, and return
  const res = await fetch(...);
  // ...
  const totalPosts = ...;
  // cache this value for 10 minutes before we need to get it again (arbitrary, could be more or less)
  const tenMinutesInSeconds = 10 * 60
  redis.set('github:totalPosts', totalPosts, 'EX', tenMinutesInSeconds);

  return totalPosts;
});

// Caller needs to deal with errors (wrap in try/catch)
const getTelescopeData = async () => ({
  totalPosts: await getTotalPosts(),
  totalFeeds: await totalFeeds(), 
  /* and so on... */
});
```

You can streamline this, but that's the idea.

Reply:
Let me experiment with it. This would be my first time using Redis.

Reply:
Added cache with redis but I'm only able to run it with  `pnpm run services:start`.
If I do `pnpm run dev` I'm getting 
```
[ioredis] Unhandled error event: Error: getaddrinfo ENOTFOUND redis
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26)
 ```

Reply:
Correct, since you have a dependency on redis now.

In your `env.local` add `MOCK_REDIS=1` to mock it in development, see https://github.com/Seneca-CDOT/satellite/blob/d42a831a8ca6666dd5759eeccdbe932fe34239ca/src/redis.js#L9